channels/smartcard/client/smartcard_operations.c

```diff
@@ -524,8 +531,12 @@ static LONG smartcard_ListReadersW_Call(SMARTCARD_DEVICE* smartcard, SMARTCARD_O
 	cchReaders = SCARD_AUTOALLOCATE;
 	status = ret.ReturnCode = SCardListReadersW(operation->hContext,
 	                          (LPCWSTR) call->mszGroups, (LPWSTR) &mszReaders, &cchReaders);
+	mszStringsLog("Readers BEFORE filtering: ", TRUE, mszReaders);
+	/* Remove the card readers that are not specified on the command line */
+	mszFilterStrings(TRUE, mszReaders, & cchReaders, smartcard->filter);
+	mszStringsLog("Readers AFTER  filtering: ", TRUE, mszReaders);
 	ret.msz = (BYTE*) mszReaders;
-	ret.cBytes = cchReaders * 2;
```

Comment:
Sure that the amount of bytes here is not times `2`? `mszReaders` is of type `LPWSTR`

Reply:
Yes, sure. cchReader is already a number of bytes.  Also, the traces show the right values..

Reply:
@informatimago Ok, could you tell the details of your setup? When I check `PCSC_SCardListReadersW` (which is called from `SCardListReadersW`) `cchReaders` is set to the number of characters just as https://docs.microsoft.com/en-us/windows/desktop/api/winscard/nf-winscard-scardlistreadersw tells too...

Reply:
( As for spaces/tab, yes, mixups are white it's better to use 100% whitespaces, and also, programmers who use whitespaces to indent tend to earn more$ ;-) )

```
find . -name \*.[hc] -exec grep -nH -e '^ \| 	' {} +|sed -e 's/:.*//'|sort -u > ../a
git diff 25e2ab1..a393bc7 | grep 'diff --git'|sed -e 's-.* b/--'|sort -u > ../b
grep -nH -e '^ \| 	' $(grep -v -F -x -f ../a ../b)
```
The only matches are 
1- lines in comments, eg. " * foo";
2- parameter list continuations.  My editor is configured to indent them with only one tab, but autoformat.sh indents them with spaces aligned to the opening parenthesis.
3- rdpdr.h in which autoformat.sh indents the enums with 4 spaces. (I let my editor indent it with tabs, but autformat.sh reverted it to spaces).

Reply:
@akallabeth PCSC_SCardListReadersW is not the last function having modified cchReaders; it's mszFilterStrings.  This one is specified to return a number of bytes in all cases (cf. str.h).